scalameta/src/main/scala/scala/meta/syntactic/Token.scala

```diff
@@ -153,3 +152,88 @@ object Token {
     def end = content.chars.length
   }
 }
+
+// NOTE: We have this unrelated code here, because of how materializeAdt works.
+//
+// Due to an unfortunate limitation of knownDirectSubclasses,
+// all macro applications that depend on that API
+// must come after the classes that they call the API on.
+//
+// That's why if we put TokenLiftables elsewhere, we might run into troubles
+// depending on how the file and its enclosing directories are called.
+// To combat that, we have TokenLiftables right here, guaranteeing that there won't be problems
+// if someone wants to refactor/rename something later.
+trait TokenLiftables extends tokens.Liftables {
```

Comment:
If I understand correctly, the redirection mechanism in `customMatcher` will see that we're trying to lift `Unquote` and use directly `liftUnquote` instead of generating a new one which would require a Liftable for `Any`? Nice!